src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -218,7 +219,7 @@ bool ArchiveBuilder::gather_klass_and_symbol(MetaspaceClosure::Ref* ref, bool re
       _klasses->append(klass);
     }
     // See RunTimeClassInfo::get_for()
-    _estimated_metaspaceobj_bytes += align_up(BytesPerWord, SharedSpaceObjectAlignment);
+    _estimated_metaspaceobj_bytes += align_up(BytesPerWord, KlassAlignmentInBytes);
```

Comment:
Small nit here: The original code was intended to reserve space for an extra pointer that's placed immediately below each Klass. I think we should do this so that the intention is clear.

```
// Reserve a slot to store a RunTimeClassInfo pointer immediately below each Klass.
// See RunTimeClassInfo::get_for()
_estimated_metaspaceobj_bytes += BytesPerWord;
// Extra padding for make sure each Klass is aligned.
_estimated_metaspaceobj_bytes += KlassAlignmentInBytes;

Reply:
Okay, I fixed it and modified the comments in both gather_klass_and_symbol and make_shallow_copy to be clearer to non-CDS people. Could you take a look at them?

Reply:
New version looks good.